### PR TITLE
RELENG-5962 set config in Vmtype object

### PR DESCRIPTION
**Pull request by tcarmet**

### CODE REVIEW

File: srcs/runners_manager/runner/RunnerFactory.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def delete_runner(self, runner: Runner):
                 self.github_manager.force_delete_runner(runner.action_id)
 
             if runner.vm_id:
-                self.cloud_manager.delete_vm(runner.vm_id, runner.vm_type.image)
+                self.cloud_manager.delete_vm(runner.vm_id, runner.vm_type.config["image"])
```
Comment (erwan-b) [inline srcs/runners_manager/runner/RunnerFactory.py:105]: Same

File: srcs/runners_manager/vm_creation/CloudManager.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import abc
 
-from marshmallow import Schema
+from marshmallow import Schema, fields
```
Comment (erwan-b) [inline srcs/runners_manager/vm_creation/CloudManager.py:3]: Please add the pre-commit config

Comment (erwan-b) [reply srcs/runners_manager/vm_creation/CloudManager.py:3]: this is not used

File: srcs/runners_manager/vm_creation/openstack/OpenstackManager.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 from runners_manager.monitoring.prometheus import metrics
 from runners_manager.runner.Runner import Runner
 from runners_manager.vm_creation.CloudManager import CloudManager
+from runners_manager.vm_creation.VmType import VmType
 from runners_manager.vm_creation.openstack.schema import OpenstackConfig
```
Comment (erwan-b) [inline srcs/runners_manager/vm_creation/openstack/OpenstackManager.py:17]: Add `VmTypeConfig` so we can check the config settings and be shure it's match our needs

File: srcs/runners_manager/runner/RunnerFactory.py [hunk 4]
```diff
@@ -81,7 +81,7 @@ def create_runner(self, vm_type: VmType):
 
     def respawn_replace(self, runner: Runner):
         logger.info(f"respawn runner: {runner.name}")
-        self.cloud_manager.delete_vm(runner.vm_id, runner.vm_type.image)
+        self.cloud_manager.delete_vm(runner.vm_id, runner.vm_type.config["image"])
```
[the remark this answers is not in the document]
Comment (tcarmet) [reply srcs/runners_manager/runner/RunnerFactory.py:84]: yes, I modified in the cloud manager the parameter needed to delete a vm with only runner, I forgot to modify the part of the code that calls those function.

File: srcs/runners_manager/vm_creation/VmType.py [hunk 5]
```diff
@@ -1,18 +1,26 @@
+
+from marshmallow import Schema, fields
+
+
+class DefaultVmConfig(Schema):
+    image = fields.Str(required=True)
+    flavor = fields.Str(required=True)
+
+
```
Comment (erwan-b) [inline srcs/runners_manager/vm_creation/VmType.py:9]: I don't like this. This is a openstack configuration template. 
We should take a dictionary in our object and have this template in the `openstack/schema.py` file.

Comment (erwan-b) [reply srcs/runners_manager/vm_creation/VmType.py:9]: Then we import this class like the `OpenstackManager` and we create a Interface like `CloudManager`, or we condiser just checking the status and then use the directly the dictionary

Comment (tcarmet) [reply srcs/runners_manager/vm_creation/VmType.py:9]: Been trying to do this today, but haven't got the time to finish yet. In progress

**Comment by tcarmet** [comment]
PR assigned to @erwan-b for now, will take over with the ticket https://scality.atlassian.net/browse/RELENG-6014

**Review by tcarmet — COMMENTED** [review]
/approve

**Review by erwan-b — APPROVED** [review]

**Pull request merged**